diff --git a/Makefile b/Makefile
index c2e6d1b..e18abb2 100644
--- a/Makefile
+++ b/Makefile
@@ -5,12 +5,12 @@
 VERSION=0.2.1
 
 VPATH=doc
-PREFIX?=/usr/local
+PREFIX?=/home/tom/src/aerc
 _INSTDIR=$(DESTDIR)$(PREFIX)
 BINDIR?=$(_INSTDIR)/bin
-SHAREDIR?=$(_INSTDIR)/share/aerc
+SHAREDIR?=/home/tom/src/aerc/config
 MANDIR?=$(_INSTDIR)/share/man
-GO?=go
+GO?=/home/tom/src/go/bin/go
 GOFLAGS?=
 
 GOSRC!=find . -name '*.go'
